c_emulator/riscv_sim.cpp

```diff
@@ -480,6 +489,9 @@ void reinit_sail(uint64_t elf_entry)
 
 void write_signature(const char *file)
 {
+  if (!sys_enable_htif()) {
```

Comment:
This is unrelated (other than that you've made load_sail exit early and not parse the signature location so have broken that combination)